src/DesktopAnalytics/Analytics.cs

```diff
@@ -63,17 +66,35 @@ public Analytics(string apiSecret, UserInfo userInfo, Dictionary<string, string>
 			_singleton = this;
 			_propertiesThatGoWithEveryEvent = propertiesThatGoWithEveryEvent;
 
-			_userInfo = userInfo;
+			if (retainPii)
```

Comment:
I like all that you're doing here, and I don't see any problems. I am nervous about the string manipulation without unit tests. If there were a bug in all of this manipulation, what is the worst that could happen?